sklearn/manifold/_t_sne.py

```diff
@@ -886,7 +896,7 @@ def _fit(self, X, skip_num_points=0):
                     distances = pairwise_distances(X, metric=self.metric, squared=True)
                 else:
                     distances = pairwise_distances(
-                        X, metric=self.metric, n_jobs=self.n_jobs
+                        X, metric=self.metric, n_jobs=self.n_jobs, **metric_params_
```

Comment:
We can move the `metric_params_` here since it is only used in this `else` statement:

```python
metric_params_ = self.metric_params or {}
distances = pairwise_distances(
    X, metric=self.metric, n_jobs=self.n_jobs, **metric_params_
)
```